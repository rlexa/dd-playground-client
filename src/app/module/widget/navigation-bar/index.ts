export * from './navigation-bar-items-from-route.directive';
export * from './navigation-bar.component';
