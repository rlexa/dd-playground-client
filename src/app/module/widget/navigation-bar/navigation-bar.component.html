<mat-toolbar color="primary">
  <span class="mat-toolbar-filler"></span>
  <ng-container *ngIf="items as iis">
    <nav class="mat-nav-list">
      <a
        *ngFor="let ii of iis; trackBy: trackByIndex"
        mat-button
        [style.min-width]="'auto'"
        [routerLink]="ii.route"
        routerLinkActive="active"
        [matTooltip]="ii.label"
      >
        <i class="material-icons transition-transform active-scale">{{ ii.icon | icon }}</i>
      </a>
    </nav>
    <span class="mat-toolbar-filler"></span>
  </ng-container>
</mat-toolbar>
