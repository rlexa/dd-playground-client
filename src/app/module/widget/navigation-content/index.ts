export * from './navigation-content.component';
