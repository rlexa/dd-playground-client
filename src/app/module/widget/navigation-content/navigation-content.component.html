<app-navigation-bar appNavigationBarItemsFromRoute layout="column"></app-navigation-bar>

<main>
  <div>
    <router-outlet></router-outlet>
  </div>
</main>
