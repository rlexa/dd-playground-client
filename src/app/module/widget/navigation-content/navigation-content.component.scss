:host {
  position: absolute;
  left: 0px;
  top: 0px;
  right: 0px;
  bottom: 0px;

  display: flex;
  flex-direction: column;

  > main {
    flex: 1 1 auto;
    position: relative;

    > div {
      position: absolute;
      left: 0px;
      top: 0px;
      right: 0px;
      bottom: 0px;

      padding: var(--size-pad);
      overflow-y: scroll;
      will-change: scroll-position, transform;
    }
  }
}
