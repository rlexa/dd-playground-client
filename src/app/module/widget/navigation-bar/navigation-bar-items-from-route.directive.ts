import {ChangeDetectorRef, Directive, OnDestroy, OnInit, inject} from '@angular/core';
import {ActivatedRoute, Data} from '@angular/router';
import {DoneSubject, RxCleanup} from 'dd-rxjs';
import {filter, map, takeUntil} from 'rxjs/operators';
import {NavigationBarComponent, NavigationBarItem} from './navigation-bar.component';

export interface NavigationBarItemsData extends Data {
  navigationBarItems: NavigationBarItem[];
}

@Directive({selector: '[appNavigationBarItemsFromRoute]', standalone: true})
export class NavigationBarItemsFromRouteDirective implements OnDestroy, OnInit {
  private readonly activatedRoute = inject(ActivatedRoute);
  private readonly navigationBarComponent = inject(NavigationBarComponent);
  private readonly changeDetectorRef = inject(ChangeDetectorRef);

  @RxCleanup() private readonly done$ = new DoneSubject();

  destroy() {}
  ngOnDestroy() {
    this.destroy();
  }

  ngOnInit() {
    this.activatedRoute.data
      .pipe(
        filter(() => !!this.navigationBarComponent),
        map((data: NavigationBarItemsData) => data?.navigationBarItems ?? null),
        takeUntil(this.done$),
      )
      .subscribe((items) => {
        this.navigationBarComponent.items = items;
        this.changeDetectorRef.markForCheck();
      });
  }
}
