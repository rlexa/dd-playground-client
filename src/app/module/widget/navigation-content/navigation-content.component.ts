import {ChangeDetectionStrategy, Component} from '@angular/core';
import {ActivatedRoute, Data} from '@angular/router';
import {map} from 'rxjs/operators';
import {NavigationBarItem} from '../navigation-bar';

export interface NavigationContentComponentRouteData extends Data {
  navs: NavigationBarItem[];
}

@Component({
  selector: 'app-navigation-content',
  templateUrl: './navigation-content.component.html',
  styleUrls: ['./navigation-content.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class NavigationContentComponent {
  constructor(private readonly activatedRoute: ActivatedRoute) {}

  readonly navs$ = this.activatedRoute.data.pipe(map((data: NavigationContentComponentRouteData) => data?.navs ?? null));
}
