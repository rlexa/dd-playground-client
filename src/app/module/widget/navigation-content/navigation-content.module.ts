import {CommonModule} from '@angular/common';
import {NgModule} from '@angular/core';
import {RouterModule} from '@angular/router';
import {NavigationBarModule} from '../navigation-bar';
import {NavigationContentComponent} from './navigation-content.component';

@NgModule({
  imports: [CommonModule, RouterModule, NavigationBarModule],
  exports: [NavigationContentComponent],
  declarations: [NavigationContentComponent],
})
class NavigationContentModule {}

export {NavigationContentModule, NavigationContentComponent};
