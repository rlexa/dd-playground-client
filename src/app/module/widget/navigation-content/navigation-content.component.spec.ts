import {ComponentFixture, TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import {MockComponents} from 'ng-mocks';
import {detectChanges, overrideForChangeDetection} from 'src/app/test';
import {NavigationBarComponent} from '../navigation-bar';
import {NavigationContentComponent} from './navigation-content.component';

describe('NavigationContentComponent', () => {
  let fixture: ComponentFixture<NavigationContentComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [NavigationContentComponent, MockComponents(NavigationBarComponent)],
      providers: [],
    })
      .overrideComponent(NavigationContentComponent, overrideForChangeDetection)
      .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(NavigationContentComponent);
    detectChanges(fixture);
  });

  it('creates instance', () => expect(fixture.componentInstance).toBeTruthy());
  it(`renders`, () => expect(fixture).toMatchSnapshot());
});
