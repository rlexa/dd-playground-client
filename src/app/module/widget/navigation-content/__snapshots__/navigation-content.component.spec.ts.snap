// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationContentComponent renders 1`] = `
<app-navigation-content
  __ngContext__={[Function LRootView]}
  activatedRoute={[Function Object]}
  scrollable$={[Function Observable]}
>
  <app-navigation-bar
    appnavigationbaritemsfromroute=""
    layout="column"
  /><main>
    <div
      class="scroll-in-y"
    >
      <router-outlet />
      
    </div>
  </main>
</app-navigation-content>
`;
